src/transformStateWithClones.js

```diff
@@ -6,8 +6,27 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  let newState = { ...state };
+  const states = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      newState = { ...newState, ...action.extraData };
+    } else if (action.type === 'removeProperties') {
+      newState = { ...newState };
+
+      for (const key of action.keysToRemove) {
+        delete newState[key];
+      }
+    } else if (action.type === 'clear') {
+      newState = {};
+    }
```

Comment:
[CODE STYLE]: Use switch statement if you have limited amount of conditions. This will make your code cleaner and more maintainable. Replace the `if...else if` chain with a `switch` statement.